src/main/java/jnr/ffi/Struct.java

```diff
@@ -101,13 +103,12 @@ protected final int addField(int sizeBits, int alignBits, Offset offset) {
         }
 
         protected final int addField(int sizeBits, int alignBits) {
-            final int off = resetIndex ? 0 : align(this.size, alignBits >> 3);
-            this.size = Math.max(this.size, off + (sizeBits >> 3));
-            this.minAlign = Math.max(this.minAlign, alignBits >> 3);
-            return off;
+            final int alignment = this.alignment.intValue() > 0 ? Math.min(this.alignment.intValue(), (alignBits >> 3)) : (alignBits >> 3);
```

Comment:
What will be if (alignBits >> 3) are less than alignment parameter? The field will align with less padding value? As I see, if the alignment parameter is set, it must be applied to any field.